### PR TITLE
fix: allows fstrings to be the first node

### LINKED ISSUE #455: ValueError: malformed node or string: _ast.JoinedStr
That's how one can break `pydocstyle` with a just 3 chars:

Reproduction:
1. `echo 'f""' > ex.py
2. `pydocstyle ex.py`

And here's the important information:

```
» pydocstyle ex.py 
Traceback (most recent call last):
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/bin/pydocstyle", line 10, in <module>
    sys.exit(main())
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/pydocstyle/cli.py", line 68, in main
    sys.exit(run_pydocstyle())
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/pydocstyle/cli.py", line 44, in run_pydocstyle
    errors.extend(check((filename,), select=checked_codes,
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 968, in check
    for error in ConventionChecker().check_source(source, filename,
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 118, in check_source
    error = this_check(self, definition,
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 158, in check_docstring_missing
    docstring and is_blank(ast.literal_eval(docstring))):
  File "/Users/sobolev/.pyenv/versions/3.8.0/lib/python3.8/ast.py", line 96, in literal_eval
    return _convert(node_or_string)
  File "/Users/sobolev/.pyenv/versions/3.8.0/lib/python3.8/ast.py", line 95, in _convert
    return _convert_signed_num(node)
  File "/Users/sobolev/.pyenv/versions/3.8.0/lib/python3.8/ast.py", line 74, in _convert_signed_num
    return _convert_num(node)
  File "/Users/sobolev/.pyenv/versions/3.8.0/lib/python3.8/ast.py", line 66, in _convert_num
    raise ValueError('malformed node or string: ' + repr(node))
ValueError: malformed node or string: <_ast.JoinedStr object at 0x10ef53610>
```

Version:

```
» pydocstyle --version
5.0.2
```

### PR DESCRIPTION
**Pull request by sobolevn**
Closes #455 

Please make sure to check for the following items:
- [x] Add unit tests and integration tests where applicable.  
      If you've added an error code or changed an error code behavior,
      you should probably add or change a test case file under `tests/test_cases/` and add 
      it to the list under `tests/test_definitions.py`.  
      If you've added or changed a command line option,
      you should probably add or change a test in `tests/test_integration.py`.
- [x] Add a line to the release notes (docs/release_notes.rst) under "Current Development Version".  
      Make sure to include the PR number after you open and get one.

### CODE REVIEW

File: src/pydocstyle/checker.py [hunk 1]
```diff
@@ -155,6 +155,7 @@ def check_docstring_missing(self, definition, docstring):
 
         """
         if (not docstring and definition.is_public or
+                docstring and docstring.lower().startswith('f') or
```
Comment (samj1912) [inline src/pydocstyle/checker.py]: Can we parenthesize these properly so that the conditions are obvious?

Comment (sobolevn) [reply src/pydocstyle/checker.py]: Sure!

**Comment by sobolevn** [comment]
Oops, the same happens with `bytes` as well: 

```
» flake8 ex.py 
Traceback (most recent call last):
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/bin/flake8", line 10, in <module>
    sys.exit(main())
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/main/cli.py", line 18, in main
    app.run(argv)
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/main/application.py", line 393, in run
    self._run(argv)
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/main/application.py", line 381, in _run
    self.run_checks()
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/main/application.py", line 300, in run_checks
    self.file_checker_manager.run()
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/checker.py", line 331, in run
    self.run_serial()
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/checker.py", line 315, in run_serial
    checker.run_checks()
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/checker.py", line 598, in run_checks
    self.run_ast_checks()
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8/checker.py", line 502, in run_ast_checks
    for (line_number, offset, text, check) in runner:
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8_docstrings.py", line 131, in run
    for error in self._check_source():
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/flake8_docstrings.py", line 112, in _check_source
    for err in self.checker.check_source(
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 118, in check_source
    error = this_check(self, definition,
  File "/Users/sobolev/Documents/github/wemake-python-styleguide/.venv/lib/python3.8/site-packages/pydocstyle/checker.py", line 179, in check_one_liners
    lines = ast.literal_eval(docstring).split('\n')
TypeError: a bytes-like object is required, not 'str'
```

I will update a PR shortly.

**Comment by sobolevn** [comment]
Also not covered: `rf` and `rb`

**Comment by samj1912** [comment]
Thanks, you might also have to check for rF/RF/Rf

**Comment by samj1912** [comment]
@sobolevn bump?

**Comment by samj1912** [comment]
Closing this as it has been inactive for a while. Please feel free to re-open if this is still relevant.

**Comment by sobolevn** [comment]
@samj1912 this PR was ready, there are other things to improve, but it can be done is separate PRs.
I don't have any time to work on this, so you can merge it or ignore.

**Comment by kstrauser** [comment]
This is still an issue in 2022-09-26. Is there a path forward to fixing it?

**Pull request closed without merging**